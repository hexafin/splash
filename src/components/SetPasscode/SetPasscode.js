import React, { Component } from "react"
import {
	View,
	Text,
	StyleSheet,
	Image,
	TouchableOpacity,
	Animated
} from "react-native"
import { colors } from "../../lib/colors"
import { defaults, icons } from "../../lib/styles"
import Keypad from '../universal/Keypad'
import ReactNativeHapticFeedback from 'react-native-haptic-feedback';
import LinearGradient from 'react-native-linear-gradient';

class SetPasscode extends Component {
	constructor(props) {
		super(props)
		this.state = {
			value: '',
			confirmValue: null,
		}
		this.onChangeText = this.onChangeText.bind(this)
		this.shakeAnimation = new Animated.Value(0)
	}

	onChangeText(char) {
		if (this.state.value.length < 4 && char != 'delete') {

			const newValue = (this.state.value+char)
			this.setState({value: newValue})
			ReactNativeHapticFeedback.trigger("impactLight", true)

			if (newValue.length == 4 && !this.state.confirmValue) {
				this.setState({confirmValue: newValue, value: ''})
			} else if (newValue.length == 4 && newValue == this.state.confirmValue) {

				this.props.navigation.state.params.successCallback(newValue)

			} else if (newValue.length == 4 && newValue != this.state.confirmValue) {
				Animated.spring(this.shakeAnimation, {
					toValue: 1,
					useNativeDriver: true,
				}).start(() => {
					this.shakeAnimation.setValue(0)
					this.setState({value: '', confirmValue: null})
				})
				ReactNativeHapticFeedback.trigger("impactHeavy", true)
			}
		} else if (char == 'delete') {
			this.setState({value: this.state.value.slice(0, -1)})
		}
	}

	render() {
		
		const dropIcon = (index) => (this.state.value.length >= index) ? <Image source={icons['whiteDrop']} style={styles.drop} resizeMode={'contain'}/>
																	   : <Image source={icons['purpleDrop']} style={styles.drop} resizeMode={'contain'}/>
		const shakeTransform =  {
			transform: [
				{
					translateX: this.shakeAnimation.interpolate({
						inputRange: [0, 0.25, 0.5, 0.75, 1],
						outputRange: [0, 40, 0, -40, 0]
					})
				}
			]
		}

		return (
			<LinearGradient colors={['#5759D5', '#4E50E6']} style={styles.container}>
				{!this.state.confirmValue && 
				<View>
					<Text style={styles.title}>Type your new passcode</Text>
					<Text style={styles.subtitle}>Splash will lock when you leave</Text>
					<Text style={[styles.subtitle, {marginBottom: 49}]}>for more than 5 minutes.</Text>
				</View>}
				{this.state.confirmValue && <Text style={styles.confirmTitle}>Confirm your passcode</Text>}
				<Animated.View style={[styles.drops, shakeTransform]}>
					{dropIcon(1)}
					{dropIcon(2)}
					{dropIcon(3)}
					{dropIcon(4)}
				</Animated.View>
				<Keypad primaryColor={'#484AD4'}
						pressColor={'#6466F6'}
						textColor={'white'}
						onChange={(char) => this.onChangeText(char)}
						disabled={(this.state.value.length >= 4) ? true : false}
						decimal={false}
						delete={true}
						arrow={'white'} />
				{/*<TouchableOpacity onPress={() => console.log('forgot')}>
					<Text style={styles.forgotText}>Forgot?</Text>
				</TouchableOpacity>*/}
			</LinearGradient>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		alignContent: 'center',
	},
	title: {
		fontSize: 20,
		fontWeight: '600',
		color: colors.white,
		paddingTop: 69,
		paddingBottom: 10,
		alignSelf: 'center'
	},
	confirmTitle: {
		fontSize: 20,
		fontWeight: '600',
		color: colors.white,
		alignSelf: 'center',
		paddingTop: 81,
		paddingBottom: 81,
	},
	subtitle: {
		fontSize: 16,
		color: '#A4A5F6',
		alignSelf: 'center',
	},
	drops: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		marginBottom: 113,
		paddingHorizontal: 105,
	},
	drop: {
		height: 15,
		width: 12,
	},
	forgotText: {
		fontSize: 14,
		color: colors.white,
		paddingTop: 38,
		alignSelf: 'center'
	},
})

export default SetPasscode
