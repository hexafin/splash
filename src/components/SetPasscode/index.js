import SetPasscode from "./SetPasscode"
import {connect} from "react-redux"
import {bindActionCreators} from "redux"

const mapStateToProps = (state) => {
    return {
    	userId: state.user.id,
    }
}

const mapDispatchToProps = (dispatch) => {
    return bindActionCreators({}, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(SetPasscode)
