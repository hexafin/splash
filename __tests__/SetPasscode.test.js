import "react-native";
import React from "react";
import SetPasscode from "../src/components/SetPasscode/SetPasscode";

it("Set Passcode renders correctly", () => {
	const snap = shallow(<SetPasscode />);
	expect(snap).toMatchSnapshot();
});
