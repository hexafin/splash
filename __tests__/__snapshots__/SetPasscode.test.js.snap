// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set Passcode renders correctly 1`] = `
<LinearGradient
  colors={
    Array [
      "#5759D5",
      "#4E50E6",
    ]
  }
  style={
    Object {
      "alignContent": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "paddingVertical": 50,
    }
  }
>
  <Component>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "flex-end",
          "paddingHorizontal": 43,
        }
      }
    >
      <Component
        source={
          Object {
            "testUri": "../../../src/assets/icons/Xbutton.png",
          }
        }
        style={
          Object {
            "height": 20,
            "width": 20,
          }
        }
      />
    </TouchableOpacity>
    <Component>
      <Component
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFFFFF",
            "fontSize": 20,
            "fontWeight": "600",
            "paddingBottom": 20,
            "paddingTop": 20,
          }
        }
      >
        Type your new passcode
      </Component>
      <Component
        style={
          Object {
            "alignSelf": "center",
            "color": "#A4A5F6",
            "fontSize": 16,
          }
        }
      >
        Splash will lock when you leave
      </Component>
      <Component
        style={
          Array [
            Object {
              "alignSelf": "center",
              "color": "#A4A5F6",
              "fontSize": 16,
            },
            Object {
              "marginBottom": 49,
            },
          ]
        }
      >
        for more than 5 minutes.
      </Component>
    </Component>
    <AnimatedComponent
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 40,
            "paddingHorizontal": 105,
            "width": 750,
          },
          Object {
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          },
        ]
      }
    >
      <Component
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/icons/purpleDrop.png",
          }
        }
        style={
          Object {
            "height": 15,
            "width": 12,
          }
        }
      />
      <Component
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/icons/purpleDrop.png",
          }
        }
        style={
          Object {
            "height": 15,
            "width": 12,
          }
        }
      />
      <Component
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/icons/purpleDrop.png",
          }
        }
        style={
          Object {
            "height": 15,
            "width": 12,
          }
        }
      />
      <Component
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/icons/purpleDrop.png",
          }
        }
        style={
          Object {
            "height": 15,
            "width": 12,
          }
        }
      />
    </AnimatedComponent>
  </Component>
  <Keypad
    arrow="white"
    decimal={false}
    delete={true}
    disabled={false}
    onChange={[Function]}
    pressColor="#6466F6"
    primaryColor="#484AD4"
    textColor="white"
  />
</LinearGradient>
`;
